### PR TITLE
Forms and buttons

### PR DESCRIPTION
**Pull request by jweakley**
New Async button and forms.

### CODE REVIEW

File: addon/components/bootstrap/-form-body.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import Ember from 'ember';
+import layout from '../../templates/components/bootstrap/-form-body';
+
+export default Ember.Component.extend({
+  layout
```
Comment (harokevin) [inline addon/components/bootstrap/-form-body.js]: is this breaking? Why is layout there?

File: addon/components/bootstrap/-form-controls.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import Ember from 'ember';
+import layout from '../../templates/components/bootstrap/-form-controls';
+
+export default Ember.Component.extend({
+  layout
```
Comment (harokevin) [inline addon/components/bootstrap/-form-controls.js]: is this breaking? Why is layout there?

File: addon/components/bootstrap/-form-body.js (hunk 1 above)
Comment (jweakley) [reply addon/components/bootstrap/-form-body.js]: addons need to specify layout.

File: addon/components/bootstrap/-form-controls.js (hunk 2 above)
Comment (jweakley) [reply addon/components/bootstrap/-form-controls.js]: Same as above

Comment (Nathan-O) [reply addon/components/bootstrap/-form-controls.js]: If I understand correctly @harokevin the `import` is only used if it is called. So this is importing the layout, then calling/applying it, even though it doesnt need to do anything except BE a layout. Someone please correct me if I'm wrong.

File: addon/components/freestyle/bootstrap-button.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  actions: {
+    simpleClick(name) {
+      return this.get('fakeSimpleClickTask').perform(name).catch(e => {
+        if (!didCancel(e)) {
+          throw e;
+        }
+      });
+    },
+    slowClick(name){
```
Comment (Nathan-O) [inline addon/components/freestyle/bootstrap-button.js]: This is just a question for my personal knowledge lack...what is the difference between a simple and slow click?

**Comment by Samsinite** [comment]
Correct Nathan, it is applying the layout defined in `/addon/templates/components/boostrap/-form-controls.hbs` hence the path in the import. Thanks for clarifying!

**Comment by chadzink** [comment]
Very cool

The Readme does not include any description for how to use this, but I am guessing this is covered by the freestyle documentation.

**Comment by jweakley** [comment]
Yep. The idea is that we are moving to live documentation with the api specs included rather than the current way which is just having the api documents. Eventually we will have controls for trying out the api of each component so that we can quickly see how components work.

File: addon/components/freestyle/bootstrap-button.js (hunk 3 above)
Comment (ike) [reply addon/components/freestyle/bootstrap-button.js]: I think this is just to demonstrate the async behavior for the button, by having one button on the Freestyle page that takes 1 second to complete its task.

**Pull request merged**